examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/DiscoveryExampleFragment.java

```diff
@@ -0,0 +1,400 @@
+  }
+
+  @Override
+  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    Log.i(TAG, "onViewCreated() called");
+
+    matterDiscoveryMessageTextView =
+        getActivity().findViewById(R.id.matterDiscoveryMessageTextView);
+    matterDiscoveryMessageTextView.setText(
+        getString(R.string.matter_discovery_message_initializing_text));
+
+    arrayAdapter = new CastingPlayerArrayAdapter(getActivity(), castingPlayerList);
+    final ListView list = getActivity().findViewById(R.id.castingPlayerList);
+    list.setAdapter(arrayAdapter);
+
+    Log.d(TAG, "onViewCreated() creating callbacks");
+
+    // TODO: In following PRs. Enable startDiscoveryButton and stopDiscoveryButton when
+    //  stopDiscovery is implemented in the core Matter SKD DNS-SD API. Enable in
+    //  fragment_matter_discovery_example.xml
+    Button startDiscoveryButton = getView().findViewById(R.id.startDiscoveryButton);
+    startDiscoveryButton.setOnClickListener(
+        v -> {
+          Log.i(
+              TAG, "onViewCreated() startDiscoveryButton button clicked. Calling startDiscovery()");
+          // arrayAdapter.clear();
+          if (!startDiscovery()) {
+            Log.e(TAG, "onViewCreated() startDiscovery() call Failed");
+          }
+        });
+
+    Button stopDiscoveryButton = getView().findViewById(R.id.stopDiscoveryButton);
+    stopDiscoveryButton.setOnClickListener(
+        v -> {
+          Log.i(TAG, "onViewCreated() stopDiscoveryButton button clicked. Calling stopDiscovery()");
+          stopDiscovery();
+          Log.i(TAG, "onViewCreated() stopDiscoveryButton button clicked. Canceling future task");
+          scheduledFutureTask.cancel(true);
+        });
+
+    Button clearDiscoveryResultsButton = getView().findViewById(R.id.clearDiscoveryResultsButton);
+    clearDiscoveryResultsButton.setOnClickListener(
+        v -> {
+          Log.i(
+              TAG, "onViewCreated() clearDiscoveryResultsButton button clicked. Clearing results");
+          arrayAdapter.clear();
+        });
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    Log.i(TAG, "onResume() called. Calling startDiscovery()");
+    if (!startDiscovery()) {
+      Log.e(TAG, "onResume() Warning: startDiscovery() call Failed");
+    }
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    Log.i(TAG, "onPause() called");
+    // stopDiscovery();
+    // Don't crash the app
+    if (scheduledFutureTask != null) {
+      scheduledFutureTask.cancel(true);
+    }
+  }
+
+  /** Interface for notifying the host. */
+  public interface Callback {
+    /** Notifies listener of Connection Button click. */
+    // TODO: In following PRs. Implement CastingPlayer connection
+    void handleConnectionButtonClicked(CastingPlayer castingPlayer);
+  }
+
+  private boolean startDiscovery() {
+    Log.i(TAG, "startDiscovery() called");
+
```

Comment:
We should clear() the CastingPlayer arrayAdapter when calling CastingPlayerDiscovery.startDiscovery()